database/sqlserver/sqlserver.go

```diff
@@ -154,7 +155,7 @@ func (ss *SQLServer) Close() error {
 
 // Lock creates an advisory local on the database to prevent multiple migrations from running at the same time.
 func (ss *SQLServer) Lock() error {
-	if ss.isLocked {
+	if ss.isLocked.Load() {
```

Comment:
Same here about fixing the local lock optimization implementation

Reply:
thnx,fixed it